applications/FluidDynamicsApplication/custom_conditions/low_mach_navier_stokes_wall_condition.cpp

```diff
@@ -0,0 +1,622 @@
+    //         auto& r_parent = this->GetValue(NEIGHBOUR_ELEMENTS)[0];
+    //         data.ViscousStress = ZeroVector(VoigtSize);
+    //         r_parent.Calculate(FLUID_STRESS, data.ViscousStress, rCurrentProcessInfo);
+    //     }
+    // }
+
+    // // Loop on gauss points
+    // for(unsigned int igauss = 0; igauss<NumGauss; igauss++)
+    // {
+    //     data.N = row(Ncontainer, igauss);
+    //     const double J = GaussPtsJDet[igauss];
+    //     data.wGauss = J * IntegrationPoints[igauss].Weight();
+    //     ComputeGaussPointRHSContribution(rhs_gauss, data, rCurrentProcessInfo);
+    //     ComputeGaussPointLHSContribution(lhs_gauss, data, rCurrentProcessInfo);
+    //     //TODO: Implement a ComputeGausPointLocalSystemContribution
+    //     noalias(rLeftHandSideMatrix) += lhs_gauss;
+    //     noalias(rRightHandSideVector) += rhs_gauss;
+    // }
+
+    // // Add the wall law contribution
+    // constexpr SizeType n_wall_models = sizeof...(TWallModel);
+    // static_assert(n_wall_models < 2, "More than one template wall model argument in 'LowMachNavierStokesWallCondition'.");
+    // if (this->Is(WALL) && n_wall_models != 0) {
+    //     (AddWallModelLocalSystemCall<TWallModel>(rLeftHandSideMatrix, rRightHandSideVector, rCurrentProcessInfo), ...);
+    // }
+
+    KRATOS_CATCH("")
+}
+
+
+template<unsigned int TDim, unsigned int TNumNodes, class... TWallModel>
+void LowMachNavierStokesWallCondition<TDim,TNumNodes,TWallModel...>::CalculateLeftHandSide(
+    MatrixType& rLeftHandSideMatrix,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Resize if needed
+    if (rLeftHandSideMatrix.size1() != LocalSize || rLeftHandSideMatrix.size2() != LocalSize) {
+        rLeftHandSideMatrix.resize(LocalSize, LocalSize, false);
+    }
+
+    // LHS contributions initialization
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(LocalSize, LocalSize);
+
+    // // Add the wall law contribution
+    // constexpr SizeType n_wall_models = sizeof...(TWallModel);
+    // static_assert(n_wall_models < 2, "More than one template wall model argument in 'LowMachNavierStokesWallCondition'.");
+    // if (this->Is(WALL) && n_wall_models != 0) {
+    //     (AddWallModelLeftHandSideCall<TWallModel>(rLeftHandSideMatrix, rCurrentProcessInfo), ...);
+    // }
+
+    KRATOS_CATCH("")
+}
+
+
+
+template<unsigned int TDim, unsigned int TNumNodes, class... TWallModel>
+void LowMachNavierStokesWallCondition<TDim,TNumNodes,TWallModel...>::CalculateRightHandSide(
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Resize if needed
+    if (rRightHandSideVector.size() != LocalSize) {
+        rRightHandSideVector.resize(LocalSize, false);
+    }
+
+    // Loop on gauss points
+    noalias(rRightHandSideVector) = ZeroVector(LocalSize);
+
+    // // Struct to pass around the data
+    // ConditionDataStruct data;
+    // // Allocate memory needed
```

Comment:
IDEM

Reply:
This is a boilerplate for the future implementation of the Neumann BCs.